src/core/binding.h

```diff
@@ -466,14 +466,14 @@ QuicRetryTokenDecrypt(
     //
     QuicCopyMemory(Token, TokenBuffer, sizeof(QUIC_RETRY_TOKEN_CONTENTS));
 
-    uint8_t Iv[QUIC_IV_LENGTH];
-    if (MsQuicLib.CidTotalLength >= sizeof(Iv)) {
-        QuicCopyMemory(Iv, Packet->DestCid, sizeof(Iv));
-        for (uint8_t i = sizeof(Iv); i < MsQuicLib.CidTotalLength; ++i) {
-            Iv[i % sizeof(Iv)] ^= Packet->DestCid[i];
+    uint8_t Iv[QUIC_MAX_IV_LENGTH];
+    if (MsQuicLib.CidTotalLength >= QUIC_IV_LENGTH) {
+        QuicCopyMemory(Iv, Packet->DestCid, QUIC_IV_LENGTH);
+        for (uint8_t i = QUIC_IV_LENGTH; i < MsQuicLib.CidTotalLength; ++i) {
+            Iv[i % QUIC_IV_LENGTH] ^= Packet->DestCid[i];
         }
     } else {
-        QuicZeroMemory(Iv, sizeof(Iv));
+        QuicZeroMemory(Iv, QUIC_IV_LENGTH);
```

Comment:
Maybe we should zero the whole thing in case the garbage is being used internally?

Reply:
Everything is working as is already. The only way we knew about this was some Windows ASAN code @maximus- was playing around with. I'd prefer to **not** add any more CPU cycles if we don't have to, especially considering crypto is the biggest bottleneck already.